compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -313,6 +321,16 @@ getMinTilingSizesForEachDim(func::FuncOp entryPointFn, linalg::LinalgOp op,
       continue;
     unsigned fastestVaryingDim = fastestVaryingDimExpr.getPosition();
 
+    // If the fastest varying dimension for the operand is broadcasted along a
+    // faster varying reduction dimension, we should prefer a vector size of 1
+    // as the values will splat along the faster varying dim.
+    if (itTypes[fastestVaryingDim] == utils::IteratorType::reduction &&
+        fastestVaryingDim < *fastestVaryingReductionDim &&
```

Comment:
That is going to assert if the value isnt set

Reply:
The value should always be set though because `fastestVaryingDim` is a reduction right?

Reply:
ok, bit convoluted here with some spooky action at distance here... Maybe guard this condition as well on `fastestVaryingRedeuctionDim` being set

Reply:
Tile size selection is not the right place to optimize a broadcast + reduction. If there are other producer operations over that dimension, they will be scalarized. We should introduce a pattern that folds a broadcast + reduction or, even better, fold this at linalg level before vectorization. Does that make sense to you?